site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -1284,8 +1287,15 @@ private function createGradeable($gradeable_id, $details) {
             );
         }
 
-        // Generate a blank component to make the rubric UI work properly
-        $this->genBlankComponent($gradeable);
+        if (isset($details['rubric'])) {
+            foreach ($details['rubric'] as $component) {
+                $gradeable->importComponent($component);
+            }
+        }
```

Comment:
A few comments about this block of code:
- Should you also check to make sure this field is iterable?
- What verifies that `$component` satisfies the contract expected by `importComponent()`?  It looks like adding additional error checking to `importComponent()` and catching the resulting exceptions would be useful here.
- What happens if there are no components?  Don't you still need to create a blank component?

Reply:
A gradeable can set it's components to a list of already created components, should I use that function instead of import components as an array? That way it wouldn't be working with "magic values" also.

Reply:
Which function are you referring to?  The idea sounds like it's on a better path at least.

Reply:
Gradeable's setComponents() function